app/controllers/teachers_controller.rb

```diff
@@ -0,0 +1,46 @@
+class TeachersController < ApplicationController
+  before_action :set_teacher, only: [:destroy]
+  before_action :set_organization
+
+  def index
+    authorize! :teach, @organization
+    @teachers = @organization.teachers
+  end
+
+  def new
+    authorize! :teach, @organization
+    @teachership = Teachership.new
+  end
+
+  def create
+    authorize! :teach, @organization
+
+    user = User.find_by_login(teacher_params[:username])
+    @teachership = Teachership.new(user: user, organization: @organization)
+
+    if @teachership.save
+      redirect_to organization_teachers_path, notice: 'User added successfully'
```

Comment:
Nit: the word user might be confusing -- `User added to <>` or maybe `Teacher added to organisation`